### DISCUSSION TITLE
The usage of the ethereum-bloom-filters library in 4.x

### DISCUSSION THREAD

**Discussion by luu-alex** [ethereum/web3.js#4443] [General]
Looked into the web3-utils validation [issue](https://github.com/chainsafe/web3.js/issues/4351) functions we should be implementing in 4.x,
For [1.x](https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-utils/src/utils.js#L27) we used the library ethereum-bloom-filters for the functions:
isBloom
isUserEthereumAddressInBloom
isContractAddressInBloom
isTopic
isTopicInBloom
isInBloom
Was thinking we can reuse the same library for these functions for 4.x , wanted to see if anyone had thoughts

**Comment by nazarhussain** [comment]
I believe `ethereum-bloom-filters` is the best package for this particular usage. We can proceed with it, unless we have any particular reason not to.

**Comment by Spacesai1or** [comment]
- [isBloom](https://github.com/joshstevens19/ethereum-bloom-filters/blob/master/src/index.ts#L7) would be simple to implement ourselves (copy/paste, no dependencies)
- [isUserEthereumAddressInBloom](https://github.com/joshstevens19/ethereum-bloom-filters/blob/master/src/index.ts#L89)
    Uses
    - `isBloom`
    - `isAddress` (included in our library)
    - `padLeft` (included in our library)
    - [isInBloom](https://github.com/joshstevens19/ethereum-bloom-filters/blob/5a3b91cecf844a493a3798b401744b95852a02a6/src/index.ts#L32)
        Uses
        - `bytesToHex` (included in our library)
        - `keccak256` (included in our library)
        - [codePointToInt](https://github.com/joshstevens19/ethereum-bloom-filters/blob/5a3b91cecf844a493a3798b401744b95852a02a6/src/index.ts#L64) (copy/paste, no dependencies)
- [isContractAddressInBloom](https://github.com/joshstevens19/ethereum-bloom-filters/blob/5a3b91cecf844a493a3798b401744b95852a02a6/src/index.ts#L118)
    Uses
    - `isBloom`
    - `isAddress`
    - `isInBloom`
- [isTopic](https://github.com/joshstevens19/ethereum-bloom-filters/blob/5a3b91cecf844a493a3798b401744b95852a02a6/src/index.ts#L155) (copy/paste, no dependencies)
- [isTopicInBloom](https://github.com/joshstevens19/ethereum-bloom-filters/blob/5a3b91cecf844a493a3798b401744b95852a02a6/src/index.ts#L139)
    Uses
    - `isBloom`
    - `isTopic`
    - `isInBloom`

---

The above shows that it would be fairly trivial to implement the needed methods ourselves. `ethereum-bloom-filters` has one dependency: [js-sha3](https://github.com/emn178/js-sha3) which hasn't been touched in **3 years**

My recommendation: Implement the needed methods ourselves, copy/pasting what we can, and integrate `ethereum-bloom-filters`'s [tests](https://github.com/joshstevens19/ethereum-bloom-filters/blob/master/src/index.spec.ts)

**Comment by jdevcs** [comment]
These functions are not complex to implement and it will reduce dependency on external lib as discussed , so sol:  Implement the needed methods ourselves.
